slf4j-api/src/main/java/org/slf4j/Logger.java

```diff
@@ -119,6 +121,14 @@ default public LoggingEventBuilder makeLoggingEventBuilder(Level level) {
      */
     public void trace(String msg);
 
+    /**
+     * Log a message at the TRACE level.
+     *
+     * @param msgSupplier supplier for the message string to be logged
+     * @since 1.4
+     */
+    public void trace(Supplier<String> msgSupplier);
```

Comment:
You have to use 
```
    default void trace(Supplier<String> msgSupplier) {
        if (isTraceEnabled()) {
            trace(msgSupplier.get());
        }
    }
```
We don't know all implementations of `Logger`.

Reply:
With default method, method "isTraceEnabled" will called twice.
I thought about it,  but we want more speed

Reply:
Yes.
We must add the default method (for backward compatibility).
We have to add the method into AbstractLogger (for performance).